### ISSUE TITLE
section starts need double newline

### ISSUE THREAD

**Issue by starpit** [probablyup/markdown-to-jsx#315]
Using 6.11.4, the following input is mishandled.

```markdown
### S1
C1
### S2
C2
```

Expected: two sections (github's own preview seems to handle this fine, as does vscode's)

Actual: The second section is not treated as a new section.

<img width="109" alt="Screen Shot 2020-06-22 at 12 24 53 PM" src="https://user-images.githubusercontent.com/4741620/85311419-68315380-b483-11ea-9ead-8e37da0f36f6.png">

whereas when i place a newline before the second section

```markdown
### S1
C1

### S2
C2
```

markdown-to-jsx parses it as expected:

<img width="50" alt="Screen Shot 2020-06-22 at 12 25 19 PM" src="https://user-images.githubusercontent.com/4741620/85311505-86974f00-b483-11ea-94e6-d79c347d29b3.png">

**Comment by czterystaczwarty** [comment]
bump on this one, just encountered it too #337

**Comment by weyert** [comment]
Yes, I am having the same issue with markdown generated via some Draft.js editor plugin

**Comment by SarKurd** [comment]
I have similar issue

```
//single newline doesn't work after the colon
test:
- li
- li


//double newline works after the colon
test:

- li
- li

```

<img width="1344" alt="Screenshot 2020-12-04 at 10 42 25" src="https://user-images.githubusercontent.com/13942717/101148164-9b286780-361d-11eb-8354-3ddf732c35ad.png">

**github markdown test which works with single newline**


//single newline works
test:
- li
- li


//double newline works
test:

- li
- li

**Comment by starpit** [comment]
fwiw, we have switched to [react-markdown](https://github.com/remarkjs/react-markdown) which, under the covers, uses remark-parse for parsing. With react-markdown v5 (which uses remark-parse v9), we have found the parsing to be pretty robust (v4/v8 had several parser bugs that impacted us, but v5/v9 seems pretty good)

**Comment by SarKurd** [comment]
@starpit thanks for suggesting, will check it out

**Comment by Vadorequest** [comment]
Using 7.1.1, this behavior hasn't changed.

**Comment by quantizor** [comment]
Should be fixed in v9
